app/src/main/java/erevacation/com/userlist/usecase/ListUC.kt

```diff
@@ -0,0 +1,33 @@
+package erevacation.com.userlist.usecase
+
+import erevacation.com.userlist.datamanager.ListDataManager
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import javax.inject.Inject
+
+class ListUC @Inject constructor(val listDManager: ListDataManager) : ListUCContract  {
+
+
+    private val TAG: String = ListUC::class.java.simpleName
+
+    private var listUCOut:ListUCContract.ListUCOut? = null
+    private val subscribeOn = Schedulers.io()
+    private val observeOn = AndroidSchedulers.mainThread()
+
+    override fun setListUCOut(ListUCOut: ListUCContract.ListUCOut) {
+       this.listUCOut = ListUCOut
+    }
+
+
+    override fun getList() {
+        listDManager.getList()
```

Comment:
We have to start using `CompositeDisposable` so we can cancel all background actions.